space/internal/components/aclnotifications/notifications.go

```diff
@@ -0,0 +1,197 @@
+			}
+			return true
+		})
+		return err
+	}
+	for _, record := range acl.Records() {
+		if err = n.handleAclContent(ctx, record, permissions, space, acl.Id()); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (n *AclNotificationSender) handleAclContent(ctx context.Context,
+	record *list.AclRecord,
+	permissions list.AclPermissions,
+	space clientspace.Space,
+	aclId string,
+) error {
+	if aclData, ok := record.Model.(*aclrecordproto.AclData); ok {
+		err := n.iterateAclContent(ctx, record, permissions, space, aclId, aclData)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (n *AclNotificationSender) iterateAclContent(ctx context.Context,
+	record *list.AclRecord,
+	permissions list.AclPermissions,
+	space clientspace.Space,
+	aclId string,
+	aclData *aclrecordproto.AclData,
+) error {
+	for _, content := range aclData.AclContent {
+		if permissions.CanManageAccounts() {
+			if reqJoin := content.GetRequestJoin(); reqJoin != nil {
+				if err := n.sendJoinRequest(ctx, reqJoin, space, record.Id, aclId); err != nil {
+					return err
+				}
+			}
+		}
+		if reqApprove := content.GetRequestAccept(); reqApprove != nil {
+			if err := n.sendParticipantRequestApprove(reqApprove, space, record.Id, aclId); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (n *AclNotificationSender) sendJoinRequest(ctx context.Context,
+	reqJoin *aclrecordproto.AclAccountRequestJoin,
+	space clientspace.Space, id, aclId string,
+) error {
+	pubKey, name, icon, err := n.getProfileData(ctx, reqJoin)
+	if err != nil {
+		return err
+	}
+	err = n.notificationService.CreateAndSend(&model.Notification{
+		Id:      id,
+		IsLocal: false,
+		Payload: &model.NotificationPayloadOfRequestToJoin{RequestToJoin: &model.NotificationRequestToJoin{
+			SpaceId:      space.Id(),
+			Identity:     pubKey.Account(),
+			IdentityName: name,
+			IdentityIcon: icon,
+		}},
+		Space: space.Id(),
+		Acl:   aclId,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (n *AclNotificationSender) sendParticipantRequestApprove(reqApprove *aclrecordproto.AclAccountRequestAccept,
+	space clientspace.Space,
+	id, aclId string,
+) error {
+	identity, _, _ := n.identityService.GetMyProfileDetails()
+	pubKey, err := crypto.UnmarshalEd25519PublicKeyProto(reqApprove.Identity)
+	if err != nil {
+		return err
+	}
+	account := pubKey.Account()
+	if account != identity {
+		return nil
+	}
+	err = n.notificationService.CreateAndSend(&model.Notification{
+		Id:      id,
+		IsLocal: false,
+		Payload: &model.NotificationPayloadOfParticipantRequestApproved{
+			ParticipantRequestApproved: &model.NotificationParticipantRequestApproved{
+				SpaceID:    space.Id(),
+				Permission: mapProtoPermissionToAcl(reqApprove.Permissions),
+			},
+		},
+		Space: space.Id(),
+		Acl:   aclId,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (n *AclNotificationSender) getProfileData(ctx context.Context, reqJoin *aclrecordproto.AclAccountRequestJoin,
```

Comment:
you could return *model.IdentityProfile, it contains pubkey.Account() already

Reply:
I suppose, `WaitProfile` function can return nil profile